pkg/ccl/streamingccl/streamingest/stream_ingestion_job.go

```diff
@@ -108,6 +108,31 @@ type streamIngestionResumer struct {
 	job *jobs.Job
 }
 
+func connectToActiveClient(
+	ctx context.Context, ingestionJob *jobs.Job,
+) (streamclient.Client, error) {
+	details := ingestionJob.Details().(jobspb.StreamIngestionDetails)
+	progress := ingestionJob.Progress()
+	topology := progress.GetStreamIngest().Topology
+
+	if len(topology.PartitionInfo) > 0 {
```

Comment:
No, I don't think I understand what's going on.  If I'm wrong about this -- please consider the sign that I still don't understand to be a strong signal that this code should be re-arranges somehow.  

Anyway, from what I can see in the current code, job Resumer loads the job, then
ingest() and ingest tries to connectToActiveClient().  I don't see what this has to do with the "aggregator hitting an error" -- as far as I can tell, this code is used on the coordinator. So, coordinator code uses topology it gets from the job record.  My confusion remains: what if the last committed topology record is entirely, and terribly wrong? Wouldn't we want to attempt refresh? 

What exactly is this caching buying us?

Reply:
It might not be clear in the PR view since its relying on unchanged code further down in the file.

The coordinator code is using `connectToActiveClient` which does use a persisted topology if one exists to read addresses, but then with that client stream_ingestion_job runs `topology, err := client.Plan(ctx, streamID)` and _that_ is the topology that's actually used for the rest of the flow.

The only real purpose of the persisted `Topology` is the array of stream addresses, the rest of it isn't read in any way, it'd only mean anything for a user/l2 looking at the value in the job table / some future helper tool we make.  

The confusion here definitely provides signal that if we're not using the rest of topology for resumption it shouldn't be in Progress.  I'll move to just using an array of stream addresses in progress and move the rest of topology to the stats function.